Packs/Vectra/Integrations/VectraEventCollector/VectraEventCollector.py

```diff
@@ -0,0 +1,438 @@
+            api_key=config.get("credentials", {}).get("password"),
+            fetch_limit=arg_to_number(arg=config.get("fetch_limit", 1000)),  # type: ignore
+            insecure=config.get("insecure"),
+            proxy=config.get("proxy"),
+        )
+
+        if cmd == "test-module":
+            result = test_module(client)
+            return_results(result)
+
+        elif cmd in ("vectra-get-events", "fetch-events"):
+
+            if cmd == "vectra-get-events":
+                should_push_events = argToBoolean(args.pop("should_push_events"))
+
+                first_fetch: datetime = arg_to_datetime(
+                    arg=config.get("first_fetch", "3 days"), arg_name="First fetch time"  # type: ignore
+                )
+                detections_cmd_res, detections, audits_cmd_res, audits = get_events(
+                    client, first_fetch
+                )
+
+                return_results(detections_cmd_res)
+                return_results(audits_cmd_res)
+
+            # fetch-events
+            else:
+
+                # We want to push events to XSIAM when fetch-events is called
+                should_push_events = True
+
+                # Not first time running fetch events
+                is_first_fetch: bool = False if demisto.getLastRun() else True
+                if not is_first_fetch:
+                    first_timestamp = demisto.getLastRun().get(DETECTION_NEXT_RUN_KEY)
+                    start = demisto.getLastRun().get(AUDIT_NEXT_RUN_KEY)
+
+                # First time running fetch events
+                else:
+                    demisto.info("First time fetching events")
+                    first_fetch: datetime = arg_to_datetime(  # type: ignore
+                        arg=config.get("first_fetch", "3 days"), arg_name="First fetch time"
+                    )
+                    first_timestamp = first_fetch.strftime(
+                        DETECTION_FIRST_TIMESTAMP_QUERY_START_FORMAT
+                    )
+
+                    start = first_fetch.strftime(AUDIT_START_TIMESTAMP_FORMAT)
+
+                detections, audits, next_fetch = fetch_events(
+                    client=client,
+                    first_timestamp=first_timestamp,
+                    start=start,
+                    is_first_fetch=is_first_fetch,
+                )
+
+                demisto.info(f"Setting last run to {str(next_fetch)}...")
+                demisto.setLastRun(next_fetch)
+
+            if should_push_events:
+                demisto.info(f"Sending {len(detections)} detections to XSIAM...")
+                send_events_to_xsiam(detections, vendor=VENDOR, product=client.endpoints[0])
+                demisto.info(f"{len(detections)} detections sent to XSIAM.")
+                demisto.info(f"Sending {len(audits)} audits to XSIAM...")
+                send_events_to_xsiam(audits, vendor=VENDOR, product=client.endpoints[1])
+                demisto.info(f"{len(audits)} audits sent to XSIAM.")
```

Comment:
Calling to send_events_to_xsiam twice is not recommended, see my comment on your [docs PR](https://github.com/demisto/content/pull/24824)

Reply:
The link is taking me back to this PR, you mean https://github.com/demisto/content-docs/pull/1300. I don't see a comment there.